requirements-testing.txt

```diff
@@ -1,7 +1,4 @@
 numpy>=1.21,<=1.23.5
 ipython>=7.34.0,<=8.12.0
 
-# docutils 0.17 introduced a bug that prevents bullets from rendering
-# see https://github.com/terrapower/armi/issues/274
-docutils <0.17
```

Comment:
This could stay removed and I could instead update the requirement in `requirements-docs.txt` (which, oddly, it has `docutils <0.18` instead)

Whatcha think @john-science?

Reply:
That's just for docs, right?

Or is it used outside the docs?

Reply:
It's used in `utils`:

![image](https://github.com/terrapower/armi/assets/5347433/c5f20229-25b3-45a5-ac25-a926ba8d924d)

But I don't think bullet point rendering matters there ;-)

It needs to be unpinned in setup.py to be python 3.11 compatible.